src/BlockTypes/FeaturedItem.php

```diff
@@ -0,0 +1,339 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use Automattic\WooCommerce\Blocks\Utils\StyleAttributesUtils;
+
+/**
+ * FeaturedItem class.
+ */
+abstract class FeaturedItem extends AbstractDynamicBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name;
+
+	/**
+	 * Default attribute values.
+	 *
+	 * @var array
+	 */
+	protected $defaults = array(
+		'align' => 'none',
+	);
+
+	/**
+	 * Global style enabled for this block.
+	 *
+	 * @var array
+	 */
+	protected $global_style_wrapper = array(
+		'text_color',
+		'font_size',
+		'border_color',
+		'border_radius',
+		'border_width',
+		'background_color',
+		'text_color',
+		'padding',
+	);
+
+	/**
+	 * Returns the featured item.
+	 *
+	 * @param array $attributes Block attributes. Default empty array.
+	 * @return \WP_Term|\WC_Product|null
+	 */
+	abstract protected function get_item( $attributes );
+
+	/**
+	 * Returns the name of the featured item.
+	 *
+	 * @param \WP_Term|\WC_Product $item Item object.
+	 * @return string
+	 */
+	abstract protected function get_item_title( $item);
+
+	/**
+	 * Returns the featured item image URL.
+	 *
+	 * @param \WP_Term|\WC_Product $item Item object.
+	 * @param string               $size    Image size, defaults to 'full'.
+	 * @return string
+	 */
+	abstract protected function get_item_image( $item, $size = 'full' );
+
+	/**
+	 * Renders the featured item attributes.
+	 *
+	 * @param \WP_Term|\WC_Product $item Item object.
+	 * @param array                $attributes Block attributes. Default empty array.
+	 * @return string
+	 */
+	abstract protected function render_attributes( $item, $attributes );
+
+	/**
+	 * Get the supports array for this block type.
+	 *
+	 * @see $this->register_block_type()
+	 * @return string[][];
+	 */
+	protected function get_block_type_supports() {
+		return array(
+			'color' => array(
+				'__experimentalDuotone' => ".wc-block-{$this->block_name}__background-image",
```

Comment:
Ok so… I have no idea why, but as far as I have seen in this repo, double quotes  are not used in favor of `.` concatenation. Personally, I really dislike that and I'd much rather have double quotes. But perhaps do you mind asking in the repo channel whether this is intentional and there is some reason behind it?

I recall asking when I first joined, here is the [very short thread](p1640364605498300-slack-C8X6Q7XQU). I think it'd be interesting to know more.